warehouse/query-core/src/main/java/datawave/query/tables/keyword/StatefulKeywordUUIDChainStrategy.java

```diff
@@ -0,0 +1,244 @@
+package datawave.query.tables.keyword;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.ListIterator;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+import java.util.UUID;
+
+import org.apache.accumulo.core.data.Key;
+import org.apache.accumulo.core.data.Value;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.log4j.Logger;
+
+import com.google.common.base.Preconditions;
+
+import datawave.microservice.query.Query;
+import datawave.microservice.query.QueryImpl;
+import datawave.query.DocumentSerialization;
+import datawave.query.QueryParameters;
+import datawave.query.attributes.Attribute;
+import datawave.query.attributes.Attributes;
+import datawave.query.attributes.Content;
+import datawave.query.attributes.Document;
+import datawave.query.attributes.TypeAttribute;
+import datawave.query.function.deserializer.DocumentDeserializer;
+import datawave.query.tables.chained.strategy.FullChainStrategy;
+import datawave.util.keyword.language.YakeLanguage;
+
+/**
+ * Strategy for chaining UUID lookup and keyword extraction queries together. Grabs the results of the lookupUUID query and uses these to generate the query
+ * terms that are fed into a keyword extraction query.
+ */
+public class StatefulKeywordUUIDChainStrategy extends FullChainStrategy<Entry<Key,Value>,Entry<Key,Value>> {
+
+    private static final Logger log = Logger.getLogger(StatefulKeywordUUIDChainStrategy.class);
+
+    private int batchSize = -1;
+    protected DocumentDeserializer deserializer;
+
+    public StatefulKeywordUUIDChainStrategy(Query settings) {
+        this.deserializer = DocumentSerialization.getDocumentDeserializer(settings);
+    }
+
+    public int getBatchSize() {
+        return batchSize;
+    }
+
+    public void setBatchSize(int batchSize) {
+        this.batchSize = batchSize;
+    }
+
+    @Override
+    protected Query buildLatterQuery(Query initialQuery, Iterator<Entry<Key,Value>> initialQueryResults, String latterLogicName) {
+        log.debug("buildLatterQuery() called...");
+
+        String queryString = captureResultsAndBuildQuery(initialQueryResults, batchSize);
```

Comment:
Need to properly handle the case where the `queryString` is empty because there were no results from the initial query. Currently, the latter query will just throw an exception that gets thrown back through the query API.

Reply:
Done